src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/NewAzureVirtualNetworkGatewayCommand.cs

```diff
@@ -162,6 +168,27 @@ public class NewAzureVirtualNetworkGatewayCommand : VirtualNetworkGatewayBaseCmd
             HelpMessage = "Do not ask for confirmation if you want to overrite a resource")]
         public SwitchParameter Force { get; set; }
 
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = VirtualNetworkGatewayParameterSets.RadiusServerConfiguration,
+            HelpMessage = "P2S External Radius server address.")]
+        [ValidateNotNullOrEmpty]
+        public string RadiusServerAddress { get; set; }
+
+        [Parameter(
```

Comment:
why would this be included in both parameter sets?

Reply:
Radius server configuration is optional - the behavior we want to have is to prompt for password if and only if user configures radius server address. Hence this is optional for the default parameter set but mandatory for the radius configuration parameter set.

Reply:
But what is the scenario for a user providng radius server configuration without a radius server address - I'm saying that this parameter should *only* be in the parameter set with the radius server address, unless I am missing some scenario

Reply:
Here is the experience we are trying to give:

1. Customer mentions everything - 
    New-AzureVirtualNetworkGateway -...... -radiusServerAddress someAddress -radiusServerSecret someSecureString ....
    
   In this case everything works.

2. Customer mentions radius server address but not radius server secret
    New-AzureVirtualNetworkGateway -...... -radiusServerAddress someAddress

  In this case, he gets prompted for the password.
  radiusServerSecret:*********

I tried multiple combinations of parameter sets - this is the combination that works for all the above scenarios.